packages/router/rollup.config.js

```diff
@@ -13,185 +11,56 @@ const {
 } = require("../../rollup.utils");
 const { name, version } = require("./package.json");
 
-module.exports = function rollup() {
+function getRollupConfig(format, filename, includeTypesAndCopy = false) {
   const { ROOT_DIR, SOURCE_DIR, OUTPUT_DIR } = getBuildDirectories(
     name,
     // We don't live in a folder matching our package name
     "router"
   );
 
-  // JS modules for bundlers
-  const modules = [
-    {
-      input: `${SOURCE_DIR}/index.ts`,
-      output: {
-        file: `${OUTPUT_DIR}/index.js`,
-        format: "esm",
-        sourcemap: !PRETTY,
-        banner: createBanner("@remix-run/router", version),
-      },
-      plugins: [
-        extensions({ extensions: [".ts"] }),
-        babel({
-          babelHelpers: "bundled",
-          exclude: /node_modules/,
-          presets: [
-            ["@babel/preset-env", { loose: true }],
-            "@babel/preset-typescript",
-          ],
-          plugins: ["babel-plugin-dev-expression"],
-          extensions: [".ts"],
-        }),
-        typescript({
-          tsconfig: path.join(__dirname, "tsconfig.json"),
-          exclude: ["__tests__"],
-          noEmitOnError: true,
-        }),
-        copy({
-          targets: [
-            { src: path.join(ROOT_DIR, "LICENSE.md"), dest: SOURCE_DIR },
-          ],
-          verbose: true,
-        }),
-      ].concat(PRETTY ? prettier({ parser: "babel" }) : []),
+  return {
+    input: `${SOURCE_DIR}/index.ts`,
+    output: {
+      file: `${OUTPUT_DIR}/${filename}`,
+      format,
+      sourcemap: !PRETTY,
+      banner: createBanner("@remix-run/router", version),
     },
-  ];
+    plugins: [
+      extensions({ extensions: [".ts"] }),
+      babel({
+        babelHelpers: "bundled",
+        exclude: /node_modules/,
+        presets: [
+          ["@babel/preset-env", { loose: true }],
+          "@babel/preset-typescript",
+        ],
+        extensions: [".ts"],
+      }),
+      ...(includeTypesAndCopy
+        ? [
+            typescript({
+              tsconfig: path.join(__dirname, "tsconfig.json"),
+              exclude: ["__tests__"],
+              noEmitOnError: true,
+            }),
+            copy({
+              targets: [
+                { src: path.join(ROOT_DIR, "LICENSE.md"), dest: SOURCE_DIR },
+              ],
+              verbose: true,
+            }),
+          ]
+        : []),
+    ].concat(PRETTY ? prettier({ parser: "babel" }) : []),
+  };
+}
 
-  // JS modules for <script type=module>
-  const webModules = [
-    {
-      input: `${SOURCE_DIR}/index.ts`,
-      output: {
-        file: `${OUTPUT_DIR}/router.development.js`,
-        format: "esm",
-        sourcemap: !PRETTY,
-        banner: createBanner("@remix-run/router", version),
-      },
-      plugins: [
-        extensions({ extensions: [".ts"] }),
-        babel({
-          babelHelpers: "bundled",
-          exclude: /node_modules/,
-          presets: ["@babel/preset-modules", "@babel/preset-typescript"],
```

Comment:
We previously had two "ESM" builds, one using `@babel/preset-env` and this one using `@babel/preset-modules`.  I went with the preset-env one to better align with our CJS build, and also because I wasn't sure if taking preset-env out of the equation would cause any node issues.  https://github.com/babel/preset-modules didn't mention anything about being targeted for node environments.